File: explorer/src/main/java/com/duongame/adapter/ExplorerListAdapter.java
package com.duongame.adapter;

import android.app.Activity;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.view.View;
import android.view.ViewGroup;

import com.duongame.R;
import com.duongame.file.FileHelper;

import java.util.ArrayList;

/**
 * Created by namjungsoo on 2016-11-16.
 */

public class ExplorerListAdapter extends ExplorerAdapter {
    public ExplorerListAdapter(Activity context, ArrayList<ExplorerItem> fileList) {
        super(context, fileList);
    }

    public void bindViewHolderExplorer(ExplorerViewHolder viewHolder, int position) {
        if (viewHolder == null)
            return;

        if(fileList == null)
            return;

        ExplorerItem item = fileList.get(position);
        if (item == null)
            return;

        viewHolder.name.setText(item.name);
        viewHolder.date.setText(item.date);
        viewHolder.size.setText(FileHelper.getMinimizedSize(item.size));
        //viewHolder.icon.setRadiusDp(5);
        viewHolder.iconSmall.setVisibility(View.INVISIBLE);
        viewHolder.position = position;
        item.position = position;

        updateCheckBox(viewHolder, item);

        setIconDefault(viewHolder, item);
        setIcon(viewHolder, item);
    }

    @Override
    public View inflateLayout(ViewGroup parent) {
        return context.getLayoutInflater().inflate(R.layout.file_list_item, parent, false);
    }

    @Override
    public int getFirstVisibleItem(RecyclerView recyclerView) {
        LinearLayoutManager layoutManager = (LinearLayoutManager) recyclerView.getLayoutManager();
        return layoutManager.findFirstVisibleItemPosition();
    }

    @Override
    public int getVisibleItemCount(RecyclerView recyclerView) {
        int first = getFirstVisibleItem(recyclerView);

        LinearLayoutManager layoutManager = (LinearLayoutManager) recyclerView.getLayoutManager();
        int last = layoutManager.findLastVisibleItemPosition();

        return last - first + 1;
    }
}

File: Explorer/src/main/java/com/duongame/adapter/ExplorerListAdapter.kt
package com.duongame.adapter

import android.app.Service
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import com.duongame.R
import com.duongame.file.FileHelper.getMinimizedSize

/**
 * Created by namjungsoo on 2016-11-16.
 */
class ExplorerListAdapter(fileList: ArrayList<ExplorerItem>) : ExplorerAdapter(fileList) {
    override fun bindViewHolderExplorer(viewHolder: ExplorerViewHolder, position: Int) {
        val item = fileList[position]
        viewHolder.name.text = item.name
        viewHolder.date.text = item.date
        viewHolder.size.text = getMinimizedSize(item.size)
        //viewHolder.icon.setRadiusDp(5);
        viewHolder.iconSmall.visibility = View.INVISIBLE
        //        viewHolder.position = position;
        item.position = position
        updateCheckBox(viewHolder, item)
        setIconDefault(viewHolder, item)
        setIcon(viewHolder, item)
    }

    override fun inflateLayout(parent: ViewGroup): View {
        return LayoutInflater.from(parent.context).inflate(R.layout.item_file_list, parent, false)
    }

    override fun getFirstVisibleItem(recyclerView: RecyclerView): Int {
        val layoutManager = recyclerView.layoutManager as LinearLayoutManager? ?: return 0
        return layoutManager.findFirstVisibleItemPosition()
    }

    override fun getVisibleItemCount(recyclerView: RecyclerView): Int {
        val first = getFirstVisibleItem(recyclerView)
        val layoutManager = recyclerView.layoutManager as LinearLayoutManager? ?: return 0
        val last = layoutManager.findLastVisibleItemPosition()
        return last - first + 1
    }
}